validator/src/types.rs

```diff
@@ -179,10 +179,16 @@ impl ValidationErrors {
     }
 
     fn add_nested(&mut self, field: &'static str, errors: ValidationErrorsKind) {
-        if let Vacant(entry) = self.0.entry(field) {
-            entry.insert(errors);
-        } else {
-            panic!("Attempt to replace non-empty ValidationErrors entry");
+        match self.0.get(field) {
+            None => {
+                self.0.insert(field, errors);
+            }
+            Some(entry) => {
+                let mut new_table = ValidationErrors::new();
+                new_table.0.insert("self", entry.clone());
+                new_table.0.insert("nested", errors);
+                self.0.insert(field, ValidationErrorsKind::Struct(Box::new(new_table)));
```

Comment:
I think rather than changing that, it would be better to redesign the whole error types so you only get 1 breaking change instead of 2 with that new change

Reply:
Would need a bit more explanation on how a redesign would cause less breaking changes. I don't see how this would cause 2 breaking changes? (I would argue that we can consider this 0 breaking changes for SemVer since all this does is fixes a panic bug case which mean the current behavior when this exact case happens is undefined?)

Reply:
Right now it panics so it doesn't work and people don't rely on it. If we merge that, people might start to rely on the self/nested fields which would be one more breaking change compared to if we did not merge that. 
I'll try to sketch what the new errors would look like this afternoon

Reply:
Oh you mean in the future if we change. Got it. I thought you meant 2 breaking changes with this PR

Reply:
@Keats what do you suggest as our path forward here?

Reply:
Looking at https://github.com/Keats/validator/pull/357, we should basically not be in a position to have both custom and nested.

In 
```
         #[validate(nested, custom(function = "all_value1s_are_unique"))] 
         a: Vec<A>, 
```
we should run the custom function first and only if it passes run the nested validations

Reply:
Sorry for delay, I think this should be done now. I have the extra case where if it is missed and we did run the nested and custom has errors, we ignore the nested too.